code/script.js

```diff
@@ -222,44 +226,57 @@ const generateBoard = () => {
   })
 }
 
+
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
 }
 
 // This function to start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  generateBoard()
+  setSecret()
+  selectQuestion()
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const value = questions.value
 
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
 }
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
   const { category, value } = currentQuestion
+  const {name, img, hair, eyes, accessories, other} = secret
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+ if (category === 'hair' || category === 'eyes') {
+   if (value === hair || value === eyes) {
+      filterCharacters(true)
+  } else {
+      filterCharacters(false)
+   }
 
-  } else if (category === 'accessories' || category === 'other') {
+  } else if (category === 'accessories') {
+      if (secret.accessories.includes(value)) {
+        filterCharacters(true)
+    } else {
+      filterCharacters(false)
+    }
 
-  }
+  } else if (category === 'other') {
+      if (secret.other.includes(value)) {
+        filterCharacters(true)
+    } else {
+      filterCharacters(false)
+    } 
+    }
 }
```

Comment:
I think the indentation is a bit of in this section, but it is difficult to say when viewing the code on github!